query/functions.go

```diff
@@ -1605,6 +1605,7 @@ func (r *FloatIntegralReducer) AggregateFloat(p *FloatPoint) {
 			r.window.end, r.window.start = r.opt.Window(p.Time)
 		}
 		r.sum = 0.0
+		r.prev = *p
```

Comment:
If we do this, then wouldn't the code below this have `r.prev` and `p` as referring to the same point? I think that would guarantee the value would be zero.

Reply:
I think that exactly was the idea, because the sum above is also set to 0.